File: 1. Practica/2. Polimorfismo/2 ejercicio/Videojuego.java
package polimorfismo;
import java.util.Scanner;
public class Videojuego {
    private String nombre;
    private String plataforma;
    private int cantidadJugadores;
    public Videojuego() {
        this.nombre = "desconocido";
        this.plataforma = "desconocida";
        this.cantidadJugadores = 0;
    }
    public Videojuego(String nombre) {
        this.nombre = nombre;
        this.plataforma = "multiplataforma";
        this.cantidadJugadores = 0;
    }
    public Videojuego(String nombre, String plataforma) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = 0;
    }
    public void agregarJugadores() {
        cantidadJugadores++;
        System.out.println("nuevo jugador total: " + cantidadJugadores);
    }
    public void agregarJugadores(int cantidad) {
        if (cantidad > 0) {
            cantidadJugadores += cantidad;
            System.out.println("se agrego " + cantidad + "  total: " + cantidadJugadores);
        } else {
            System.out.println("error...");
        }
    }
    public void agregarJugadores(Scanner scanner) {
        System.out.print("cantidad de jugadores a agregar: ");
        int cantidad = scanner.nextInt();
        agregarJugadores(cantidad);
    }
    public void mostrarDatos() {
        System.out.println("nombre: " + nombre);
        System.out.println("plataforma: " + plataforma);
        System.out.println("cantidad de jugadores: " + cantidadJugadores);
    }
}

File: 1. Practica/2. Polimorfismo/2 ejercicio/videojuego.py
from multipledispatch import dispatch

class Videojuego:
    def __init__(self, nombre="desconocido", plataforma="desconocida"):
        self.nombre = nombre
        self.plataforma = plataforma
        self.cantidadJugadores = 0
    @dispatch()
    def agregarJugadores(self):
        self.cantidadJugadores += 1
        print(f"Nuevo jugador total: {self.cantidadJugadores}")
    @dispatch(int)
    def agregarJugadores(self, cantidad):
        if cantidad > 0:
            self.cantidadJugadores += cantidad
            print(f"Se agregaron {cantidad}, total: {self.cantidadJugadores}")
        else:
            print("Error: cantidad inválida")
    @dispatch(object)
    def agregarJugadores(self, scanner):
        cantidad = int(input("Cantidad de jugadores a agregar: "))
        self.agregarJugadores(cantidad)

    def mostrarDatos(self):
        print(f"Nombre: {self.nombre}")
        print(f"Plataforma: {self.plataforma}")
        print(f"Cantidad de jugadores: {self.cantidadJugadores}")

j1 = Videojuego()
j2 = Videojuego("FIFA 2025")
j3 = Videojuego("tarza", "play")
j1.agregarJugadores()
j2.agregarJugadores(3)
j1.mostrarDatos()
j2.mostrarDatos()
j3.mostrarDatos()
